neuro_config_client/factories.py

```diff
@@ -814,3 +828,92 @@ def create_ingress(cls, ingress: IngressConfig) -> dict[str, Any]:
         if ingress.cors_origins:
             result["cors_origins"] = ingress.cors_origins
         return result
+
+    @classmethod
+    def create_node_pool(
+        cls, node_pool: NodePool, cloud_provider_type: CloudProviderType
+    ) -> dict[str, Any]:
+        if cloud_provider_type == CloudProviderType.ON_PREM:
+            return cls._create_onprem_node_pool(node_pool)
+        elif cloud_provider_type.value.startswith("vcd_"):
+            return cls._create_vcd_node_pool(node_pool)
+        else:
+            return cls._create_public_cloud_node_pool(node_pool, cloud_provider_type)
+
+    @staticmethod
+    def _create_onprem_node_pool(node_pool: NodePool) -> dict[str, Any]:
+        result = {
+            "name": node_pool.name,
+            "role": node_pool.role.value,
+            "min_size": node_pool.min_size,
+            "max_size": node_pool.max_size,
+            "idle_size": node_pool.idle_size,
+            "cpu": node_pool.cpu,
+            "available_cpu": node_pool.available_cpu,
+            "memory_mb": node_pool.memory_mb,
+            "available_memory_mb": node_pool.available_memory_mb,
+            "disk_size_gb": node_pool.disk_size_gb,
+            "price": str(node_pool.price),
+            "is_preemptible": node_pool.is_preemptible,
+            "zones": node_pool.zones,
+        }
+        if node_pool.machine_type:
+            result["machine_type"] = node_pool.machine_type
+        if node_pool.disk_type:
+            result["disk_type"] = node_pool.disk_type
+        if node_pool.gpu:
+            result["gpu"] = node_pool.gpu
+        if node_pool.gpu_model:
+            result["gpu_model"] = node_pool.gpu_model
+        if node_pool.currency:
+            result["currency"] = node_pool.currency
+        if node_pool.zones:
+            result["zones"] = node_pool.zones
```

Comment:
Server validator doesn't accept zones

Reply:
removed in [1972938](https://github.com/neuro-inc/neuro-config-client/pull/83/commits/1972938fc1d597d4fb0cbd449776a344b17e53a2)